### DISCUSSION TITLE
Help with using accessToken in Octokit

### DISCUSSION THREAD

**Discussion by lokmanm** [nextauthjs/next-auth#1705] [Help] [answered]
Hey everyone

I added the Github access token to my session with the following code:

```js
  callbacks: {
    async jwt(token, user, account, profile, isNewUser) {
      // Add access_token to the token right after signin
      if (account?.accessToken) {
        token.accessToken = account.accessToken
        token.username = profile.login
      }

      return token
    },

    async session(session, token) {
      // Add property to session, like an access_token from a provider.
      session.accessToken = token.accessToken
      session.user.username = token.username

      return session
    },
  },
```

But now whenever I try to use the octokit for an authorized API call I get a 404.

Example code:
```js
  const session = await getSession({ req });

  const octokit = new Octokit({
    auth: session.accessToken
  });


  const { data } = await octokit.request("GET /orgs/{org}/issues", {
    org: 'private-repo-here',
    filter: 'assigned',
    state: 'open',
  });
```

Has anyone had a similar issue ? I tried a lot of combinations but I cant get it to work. Thanks

**Comment by balazsorban44** [comment, marked as answer]
404 means you are request a resource from a wrong url. I'm not familiar with octokit, but try to re-verify if you are using the correct url again. sorry if that is obvious, I don't have a better suggestion at the moment

**Reply by lokmanm** [reply]
Hey @balazsorban44, actually they return 404 for every request ( unauthorized, not found, etc )  because of security reasons. I actually took the example from their docs website but its not working. If I try with a Personal Access Token ( one generated from Settings ) same url works just fine.

**Reply by balazsorban44** [reply]
Strange, if it is an auth reason, it really should be something like 401 as a response...

But did you add the correct scopes to the GitHub provider?

**Reply by lokmanm** [reply]
This is how my nextauth file looks like. Signin and retrieving user info works just fine.

```js
import NextAuth from 'next-auth'
import Providers from 'next-auth/providers'

export default NextAuth({
  providers: [
    Providers.GitHub({
      clientId: process.env.GITHUB_ID,
      clientSecret: process.env.GITHUB_SECRET,
    }),
  ],
  callbacks: {
    async jwt(token, user, account, profile, isNewUser) {
      // Add access_token to the token right after signin
      if (account?.accessToken) {
        token.accessToken = account.accessToken
        token.username = profile.login
      }

      return token
    },

    async session(session, token) {
      // Add property to session, like an access_token from a provider.
      session.accessToken = token.accessToken
      session.user.username = token.username

      return session
    },
  },
})
```

**Reply by balazsorban44** [reply]
Yes, could you make sure that you have all the needed scopes for GitHub and the action you are trying to do?

By default, it is only `user`

https://github.com/nextauthjs/next-auth/blob/main/src/providers/github.js

**Reply by lokmanm** [reply]
Yep that one was it, sorry I had no idea an option like that exists. Spent the whole day looking at the documentation but missed this one. 

Thanks a ton !

**Reply by balazsorban44** [reply]
Cheers! Yeah, I think we should make it more obvious that all the options in a custom provider CAN be extended on the built-in ones as well. Once you understand it it's so powerful!
